packages/core/src/test/awsService/cloudWatchLogs/registry/liveTailSession.test.ts

```diff
@@ -0,0 +1,84 @@
+    const testLogGroupArn = 'arn:aws:test-log-group'
+    const testRegion = 'test-region'
+    const testFilter = 'test-filter'
+    const testAwsCredentials = {} as any as AWS.Credentials
+
+    it('builds StartLiveTailCommand: no stream Filter, no event filter.', function () {
+        const session = buildLiveTailSession({ type: 'all' }, undefined)
+        validateStartLiveTailCommand(
+            session.buildStartLiveTailCommand(),
+            new StartLiveTailCommand({
+                logGroupIdentifiers: [testLogGroupArn],
+                logEventFilterPattern: undefined,
+                logStreamNamePrefixes: undefined,
+                logStreamNames: undefined,
+            })
+        )
+    })
+
+    it('builds StartLiveTailCommand: with prefix stream Filter', function () {
+        const session = buildLiveTailSession({ type: 'prefix', filter: testFilter }, undefined)
+        validateStartLiveTailCommand(
+            session.buildStartLiveTailCommand(),
+            new StartLiveTailCommand({
+                logGroupIdentifiers: [testLogGroupArn],
+                logEventFilterPattern: undefined,
+                logStreamNamePrefixes: [testFilter],
+                logStreamNames: undefined,
+            })
+        )
+    })
+
+    it('builds StartLiveTailCommand: with specific stream Filter', function () {
+        const session = buildLiveTailSession({ type: 'specific', filter: testFilter }, undefined)
+        validateStartLiveTailCommand(
+            session.buildStartLiveTailCommand(),
+            new StartLiveTailCommand({
+                logGroupIdentifiers: [testLogGroupArn],
+                logEventFilterPattern: undefined,
+                logStreamNamePrefixes: undefined,
+                logStreamNames: [testFilter],
+            })
+        )
+    })
+
+    it('builds StartLiveTailCommand: with log event Filter', function () {
+        const session = buildLiveTailSession({ type: 'all' }, testFilter)
+        validateStartLiveTailCommand(
+            session.buildStartLiveTailCommand(),
+            new StartLiveTailCommand({
+                logGroupIdentifiers: [testLogGroupArn],
+                logEventFilterPattern: testFilter,
+                logStreamNamePrefixes: undefined,
+                logStreamNames: undefined,
+            })
+        )
+    })
+
+    function buildLiveTailSession(
+        logStreamFilter: LogStreamFilterResponse,
+        logEventFilterPattern: string | undefined
+    ): LiveTailSession {
+        return new LiveTailSession({
+            logGroupArn: testLogGroupArn,
+            logStreamFilter: logStreamFilter,
+            logEventFilterPattern: logEventFilterPattern,
+            region: testRegion,
+            awsCredentials: testAwsCredentials,
+        })
+    }
+
+    function validateStartLiveTailCommand(actual: StartLiveTailCommand, expected: StartLiveTailCommand) {
+        assert.strictEqual(JSON.stringify(actual), JSON.stringify(expected))
```

Comment:
instead of json.stringify, any reason not to use  `deepStrictEqual` , which will also produce more-readable test failures? Also, this would eliminate the need for this wrapper function.

```suggestion
        assert.deepStrictEqual(actual, expected)
```

Reply:
This hadn't been working for me. I think there are some properties in the Command object that are generated outside of the input provided that doesn't match (something in the middlewares attached to the command)

But I just learned we can access the `.input` property of the command to get back just the request parameters. I can assert.deepStrictEqual on this! This is what we are trying to test here anyways, the request parameters.